src/Phinx/Db/Adapter/PdoAdapter.php

```diff
@@ -292,17 +292,68 @@ public function disconnect()
     /**
      * {@inheritdoc}
      */
-    public function execute($sql)
+    public function execute($sql, array $inputParams = null)
     {
-        return $this->getConnection()->exec($sql);
+        if (empty($inputParams)) {
+            $rowCount = $this->getConnection()->exec($sql);
+        } else {
+            $rowCount = $this->executeParameterisedQuery($sql, $inputParams)->rowCount();
+        }
+
+        return $rowCount !== false;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function query($sql)
+    public function query($sql, array $inputParams = null)
     {
-        return $this->getConnection()->query($sql);
+        if (empty($inputParams)) {
+            $statement = $this->getConnection()->query($sql);
+        } else {
+            $statement = $this->executeParameterisedQuery($sql, $inputParams);
+        }
+
+        return $statement;
+    }
+
+
+    /**
+     * @param $sql
+     * @param array $inputParams
+     * @return \PDOStatement
+     */
+    protected function executeParameterisedQuery($sql, array $inputParams)
+    {
+        $statement = $this->getConnection()->prepare($sql);
+
+        $position = 1;
+        foreach ($inputParams as $key => $bind) {
+            $value = $bind;
+            $type = \PDO::PARAM_STR;    // This is PDO's default type already.
```

Comment:
Hey eridal,

PDO will always use \PDO::PARAM_STR if you don't otherwise specify:

    PDOStatement::bindValue ( mixed $parameter , mixed $value [, int $data_type = PDO::PARAM_STR ] )

I don't mind adopting your suggestion, but the outcome is the same.  Let me know.